src/api/docs/content/specs/localdns.yaml

```
@@ -0,0 +1,199 @@
+          '401':
+            description: Unauthorized
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: 'common.yaml#/components/errors/unauthorized'
+                    - $ref: 'common.yaml#/components/schemas/took'
+          '404':
+            description: Item not found
+            content:
+              application/json:
+                schema:
+                  $ref: 'common.yaml#/components/schemas/took'
+
+    cname_value:
+      parameters:
+        - $ref: 'localdns.yaml#/components/parameters/cname_value'
+      put:
+        summary: Add a CNAME record
+        tags:
+          - "Local DNS Records"
+        operationId: "add_cname_value"
+        description: |
+          Add a CNAME DNS record to the Pi-hole dnsmaq.conf file.
+        responses:
+          '201':
+            description: Created item
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: 'common.yaml#/components/schemas/took'
+          '400':
+            description: Bad request
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: 'common.yaml#/components/errors/bad_request'
+                    - $ref: 'common.yaml#/components/schemas/took'
+          '401':
+            description: Unauthorized
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: 'common.yaml#/components/errors/unauthorized'
+                    - $ref: 'common.yaml#/components/schemas/took'
+      delete:
+        summary: Delete a CNAME record
+        tags:
+          - "Local DNS Records"
+        operationId: "delete_cname_value"
+        description: |
+          Delete a local 'CNAME' DNS record
+        responses:
+          '204':
+            description: No Content (deleted)
+          '401':
+            description: Unauthorized
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: 'common.yaml#/components/errors/unauthorized'
+                    - $ref: 'common.yaml#/components/schemas/took'
+          '404':
+            description: Item not found
+            content:
+              application/json:
+                schema:
+                  $ref: 'common.yaml#/components/schemas/took'
+
+  parameters:
+    record_type:
+      in: path
+      name: type
+      schema:
+        type: string
+        enum:
+          - hosts
+          - cnameRecords
+      required: true
+      description: The type of DNS record hosts (A) or cnameRecord (CNAME)
+      example: hosts
+    host_value:
+      in: path
+      name: record
+      schema:
+        type: string
+      required: true
+      description: The IP address (IPv4 or IPv6) and domain seperated by a 'SPACE'
```

There is a small typo here: `seperated ==> separated`

```suggestion
      description: The IP address (IPv4 or IPv6) and domain separated by a 'SPACE'
```